File: bench/go/tripleloop.go
// bench/go/tripleloop.go — Go 版三层嵌套循环（与 bench/myp/tripleloop.myp 同算法）
// verify <累加和>；与 C++/MYP 同算法。
package main

import (
	"fmt"
	"time"
)

func tripleLoop(n int) int64 {
	var sum int64
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			for k := 0; k < n; k++ {
				sum += int64((i*101 + j*97 + k*89) % 997)
			}
		}
	}
	return sum
}

func main() {
	const n = 300
	t0 := time.Now()
	s := tripleLoop(n)
	ms := time.Since(t0).Milliseconds()
	fmt.Printf("verify %d\nms %d\n", s, ms)
}

File: bench/cpp/tripleloop.cpp
// bench/cpp/tripleloop.cpp — C++ 版三层嵌套循环（与 bench/myp/tripleloop.myp 同算法）
// 输出：verify <累加和> / ms <毫秒>
#include <chrono>
#include <cstdio>

static long tripleLoop(int n) {
    long sum = 0;
    for (int i = 0; i < n; ++i)
        for (int j = 0; j < n; ++j)
            for (int k = 0; k < n; ++k)
                sum += (i * 101 + j * 97 + k * 89) % 997;
    return sum;
}

int main() {
    const int n = 300;
    auto t0 = std::chrono::steady_clock::now();
    long s = tripleLoop(n);
    auto t1 = std::chrono::steady_clock::now();
    double ms = std::chrono::duration<double, std::milli>(t1 - t0).count();
    std::printf("verify %ld\nms %.0f\n", s, ms);
    return 0;
}
